Mathlib/Analysis/Complex/UpperHalfPlane/Metric.lean

```diff
@@ -384,4 +384,47 @@ instance : IsometricSMul SL(2, ℝ) ℍ :=
       exact
         (isometry_real_vadd w).comp (h₀.comp <| (isometry_real_vadd v).comp <| isometry_pos_mul u)⟩
 
+section slices
+
+/--The verticel strip of width A and height B-/
+def upperHalfPlaneSlice (A B : ℝ) :=
+  {z : ℍ | Complex.abs z.1.1 ≤ A ∧ Complex.abs z.1.2 ≥ B}
+
+theorem slice_mem_iff (A B : ℝ) (z : ℍ) :
+    z ∈ upperHalfPlaneSlice A B ↔ Complex.abs z.1.1 ≤ A ∧ Complex.abs z.1.2 ≥ B := Iff.rfl
+
+lemma subset_slice_of_isCompact {K : Set ℍ} (hK : IsCompact K) : ∃  A B : ℝ, 0 < B ∧
+    K ⊆ upperHalfPlaneSlice A B  := by
+  by_cases hne : Set.Nonempty K
```

Comment:
Maybe start with `obtain rfl | hne := Set.eq_empty_or_nonempty K`, and then you can de-indent the main part of the proof